tasks/elastalert.yml

```diff
@@ -30,10 +30,20 @@
   become_user: "{{ elastalert_user }}"
   become: yes
 
+- name: Empty up the rules directory
+  file:
+    state: absent
+    path: "{{ elastalert_etc_root }}/{{ elastalert_rules_dir }}/"
+
+- name: Ensure rules directory exists
+  file:
+    state: directory
```

Comment:
this logic is going to be problematic because it will break idempotency. The directory will always get recursively removed above with the `absent` attribute and then restored and files copied back in. I see two ways to work around your issue:
- over-load the `elastalert_rules_dir` variable to select a naming convention that doesnt interact with a pre-existing file (easier short term fix)
- fix the logic to detect if that `elastalert_rules_dir` path is really a file, if so, back it up, and then create the directory accordingly.

Reply:
I don't understand the idempotency part. It will produce the same result always. 
The new  approach **does** break backwards compatibility because `elastalert_rules_dir` was not a directory per se.

How about this approach:
- Create a new variable `separate_file_per_rule: false`
- This step `Empty up the rules directory` will only run when `separate_file_per_rule` is `true`
- Modify the `Copy over alert files` accordingly

Reply:
To clarify re: idempotency,  the issue i see is that the code now has a state of `absent` directly followed by a `directory` state. On re-runs of the role, that logic will always trigger a wipe followed by laying files down - and you will always get a report from ansible indicating that tasks have changed.

What if we just fail in the scenario that the `elastalert_rules_dir ` exists and is not in-fact  a directory? Then we leave it up to the admin to either modify that path or they can over-ride the `elastalert_rules_dir` (since its an ansible variable) and point it to a new location?

Reply:
That makes sense. Let me craft a proposal